File: 2term.labs/lab 2.4/c#/Log.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ClassLibrary1
{
    internal class Log
    {
        private double Num { get; }
        public Log(double num)
        {
            Num = num;
        }

        public double Calculate()
        {
            if (Num <= 0)
            {
                throw new ArgumentException("Error. Argument for lg must be > 0!");
            }
            double result = Math.Log10(Num);

            return result;
        }

    }
}

File: 2term.labs/lab 2.4/c++/Log.cpp
#include "Log.h"
#include <iostream>
#include <cmath>
#include <string>
using namespace std;

Log::Log(double num1) : num(num1) {};

double Log::calculate() {
	if (num <= 0) {
		string str = "Error. Argument for lg must be > 0!";
		throw str;
	}
	return log10(num);
}